libretroshare/src/retroshare/rsservicecontrol.h

```diff
@@ -96,26 +112,95 @@ class RsServicePermissions
     // if DefaultAllowed = false, then only PeersAllowed is checked.
     std::set<RsPeerId> mPeersAllowed;
     std::set<RsPeerId> mPeersDenied;
+
+	// RsSerializable interface
+public:
+	void serial_process(RsGenericSerializer::SerializeJob j, RsGenericSerializer::SerializeContext &ctx) {
+		RS_SERIAL_PROCESS(mServiceId);
+		RS_SERIAL_PROCESS(mServiceName);
+
+		RS_SERIAL_PROCESS(mDefaultAllowed);
+
+		RS_SERIAL_PROCESS(mPeersAllowed);
+		RS_SERIAL_PROCESS(mPeersDenied);
+	}
 };
 
 class RsServiceControl
 {
-	public:
+public:
 
 	RsServiceControl()  { return; }
-virtual ~RsServiceControl() { return; }
-
-virtual bool getOwnServices(RsPeerServiceInfo &info) = 0;
-virtual std::string getServiceName(uint32_t service_id) = 0;
-virtual bool getServiceItemNames(uint32_t service_id,std::map<uint8_t,std::string>& names) = 0;
-
-virtual bool getServicesAllowed(const RsPeerId &peerId, RsPeerServiceInfo &info) = 0;
-virtual bool getServicesProvided(const RsPeerId &peerId, RsPeerServiceInfo &info) = 0;
-virtual bool getServicePermissions(uint32_t serviceId, RsServicePermissions &permissions) = 0;
-virtual bool updateServicePermissions(uint32_t serviceId, const RsServicePermissions &permissions) = 0;
-
-virtual void getPeersConnected(const uint32_t serviceId, std::set<RsPeerId> &peerSet) = 0;
-
+	virtual ~RsServiceControl() { return; }
```

Comment:
now that you edited the line please take the chance to remove the horrible `return;` on the empty method